server/routes/chefs.js

```diff
@@ -0,0 +1,119 @@
+        res.status(500).json({ errors: ["Unexpected error occured"] });
+    }
+});
+
+// get the chef profile for signedin user
+router.get("/", async (req, res) => {
+    try {
+        const { id } = req.user;
+        const chef = await chefsController.findOneWithUserId(id);
+        if (!chef) {
+            res.status(400).json({
+                errors: ["Chef profile not found for loggedin user"],
+            });
+            return;
+        }
+        res.json(chef);
+    } catch (error) {
+        console.error(error);
+        res.status(500).json({ errors: ["Unexpected error occured"] });
+    }
+});
+
+// update the chef profile for signedin user
+router.put("/", async (req, res) => {
+    try {
+        const { id } = req.user;
+        let { cuisineSpecialty } = req.body;
+        const chef = await chefsController.findOneWithUserId(id);
+        if (!chef) {
+            res.status(400).json({
+                errors: ["Chef profile not found for loggedin user"],
+            });
+            return;
+        }
+        const errors = [];
+        if (!cuisineSpecialty) {
+            errors.push("Missing cuisineSpecialty");
+        } else {
+            cuisineSpecialty = JSON.parse(cuisineSpecialty);
+            if (!isArrayOfStrings(cuisineSpecialty))
+                errors.push("Invalid cuisineSpecialty type");
+        }
+
+        if (errors.length > 0) {
+            res.status(400).json({ errors });
+            return;
+        }
+        const newChef = await chefsController.update(chef, {
+            cuisineSpecialty,
+            userId: id,
+        });
+        res.json(newChef);
+    } catch (error) {
+        console.error(error);
+        res.status(500).json({ errors: ["Unexpected error occured"] });
+    }
+});
+
+// create a chef profile for signedin user
+router.post("/", async (req, res) => {
+    try {
+        const { id } = req.user;
+        let { cuisineSpecialty } = req.body;
+        const chef = await chefsController.findOneWithUserId(id);
+        if (chef) {
+            res.status(400).json({
+                errors: ["Chef already exist for loggedin user"],
+            });
+            return;
+        }
+        const errors = [];
+        if (!cuisineSpecialty) {
+            errors.push("Missing cuisineSpecialty");
+        } else {
+            cuisineSpecialty = JSON.parse(cuisineSpecialty);
+            if (!isArrayOfStrings(cuisineSpecialty))
+                errors.push("Invalid cuisineSpecialty type");
+        }
+
+        if (errors.length > 0) {
+            res.status(400).json({ errors });
+            return;
+        }
+        const newChef = await chefsController.create({
+            cuisineSpecialty,
+            userId: id,
+        });
+        await usersController.update(id, { isChef: true });
+        res.json(newChef);
+    } catch (error) {
+        console.error(error);
+        res.status(500).json({ errors: ["Unexpected error occured"] });
+    }
+});
+
```

Comment:
There's a lot of repetition in this file. Can we break some of it out into reusable functions to DRY it up?

Reply:
fixed!